src/components/CompareResults/beta/ResultsView.tsx

```diff
@@ -39,9 +43,13 @@ function ResultsView(props: ResultsViewProps) {
     const searchParams = new URLSearchParams(location.search);
     const repos = searchParams.get('repos')?.split(',');
     const revs = searchParams.get('revs')?.split(',');
-    void dispatchFetchCompareResults(repos as Repository['name'][], revs);
 
     if (revs && repos) {
+      dispatch(clearSelectedRevisions());
+
+      void dispatchFetchCompareResults(repos as Repository['name'][], revs);
+
+      //On component mount, use the repos and revs in hash to search for the base and new revisions. Store the results in state via the SelectedRevisionsSlice: see extra reducer, fetchRevisionsByID. Now can always display the selected revisions despite page refresh or copying and pasting url
```

Comment:
nit: please split this on several lines.
(I think prettier would do it, but without prettier you can do it yourself :) )